adminSiteServer/FunctionalRouter.ts

```diff
@@ -57,4 +60,11 @@ export class FunctionalRouter {
     ) {
         this.router.delete(targetPath, this.wrap(callback))
     }
+
+    postWithFileUpload(
+        targetPath: string,
+        callback: (req: Request, res: Response) => Promise<any>
+    ) {
+        this.router.post(targetPath, upload.single("file"), this.wrap(callback))
```

Comment:
nice! And if we ever want to move to multiple uploads, it's nice to know that the change on multer's side is straightforward